app/services/stat_retrieval_service.rb

```diff
@@ -1,43 +1,52 @@
 class StatRetrievalService
-  def self.find(user)
-    user_stats = {
-      win_ratio: user.win_ratio,
-      wins: user.games_won,
-      losses: user.games_lost
-    }
-
-    return user_stats unless user.ranked?
-
-    user_stats[:rank] = rankings.find { |user_stats| user_stats[:slack_name] == user.slack_user_name }[:rank]
-    user_stats
-  end
+  class << self
+    def find(user)
+      wins = user.games_won
+      losses = user.games_lost
+      {
+        rank: user.rank,
+        win_ratio: User.calculate_win_ratio(wins, wins + losses),
+        wins: wins,
+        losses: losses
+      }
+    end
 
-  def self.rankings
-    sorted_win_ratios = []
-    User.all.each do |user|
-      next unless user.ranked?
-      sorted_win_ratios.push({ win_ratio: user.win_ratio, slack_name: user.slack_user_name, wins: user.games_won, losses: user.games_lost })
+    def stats_string(user)
+      user_stats = find(user)
+      ["Elo Rank: #{user_stats[:rank]}",
+       "Wins: #{user_stats[:wins]}\tLosses: #{user_stats[:losses]}",
+       "#{user} has won #{user_stats[:win_ratio]}% of the games they have finished."]
+        .join("\n")
     end
-    sorted_win_ratios.sort! { |x,y| y[:win_ratio] <=> x[:win_ratio] }
-    determine_ranks(sorted_win_ratios)
-  end
 
-  ####PRIVATE METHODS####
+    def rankings
+      ranked_users = []
+      User.order('rank desc').each do |u|
+        next unless u.ranked?
+        wins = u.games_won
+        losses = u.games_lost
+        win_ratio = User.calculate_win_ratio(wins, wins + losses)
+
+        ranked_users << { slack_name: u.slack_user_name,
+                          rank: u.rank,
+                          wins: wins,
+                          losses: losses,
+                          win_ratio: win_ratio }
 
-  def self.determine_ranks(sorted_user_win_ratios)
-    previous_win_ratio = -1
-    current_rank = 0
-    players_at_rank = 1
-    sorted_user_win_ratios.each do |user_win_ratio_hash|
-      if user_win_ratio_hash[:win_ratio] != previous_win_ratio
-        current_rank += players_at_rank
-        players_at_rank = 0
-        previous_win_ratio = user_win_ratio_hash[:win_ratio]
+        return ranked_users if ranked_users.size > 20
```

Comment:
This method could return all the rankings by default, and this limit of `21` could be passed in via `rankings(limit: 21)`. You could also use this in the original query, i.e. `User.order('rank desc').limit(limit).each do |u|`. To get all rankings by default, just set the default limit to a really large number (see http://dev.mysql.com/doc/refman/5.5/en/select.html, search for `18446744073709551615`).

This would let us control the number of users displayed in the slack rankings list independently of the ranking list shown in the web UI, which might be nice.